snakemake/executors/touch.py

```diff
@@ -30,22 +30,48 @@
 
 
 class Executor(RealExecutor):
+    SLEEPING_TIME = 0.1
+
     def run_job(
         self,
         job: JobExecutorInterface,
     ):
         job_info = SubmittedJobInfo(job=job)
         try:
-            time.sleep(0.1)
 
             if job.output:
 
                 async def touch():
-                    for f in job.output:
-                        if f.is_storage and await f.exists_in_storage():
+                    touch_storage_and_local_files = set(
+                        [
+                            f
+                            for f in job.output
+                            if f.is_storage and await f.exists_in_storage()
+                        ]
+                    )
+                    touch_files = set(
+                        [
+                            f
+                            for f in job.output
+                            if f not in touch_storage_and_local_files
+                            and await f.exists_local()
+                        ]
+                    )
+                    non_existing_files = (
+                        set(job.output) - touch_storage_and_local_files - touch_files
+                    )
+
+                    if (len(touch_files) + len(touch_storage_and_local_files)) > 0:
+                        time.sleep(Executor.SLEEPING_TIME)
+                        for f in touch_storage_and_local_files:
                             await f.touch_storage_and_local()
-                        elif await f.exists_local():
+                        for f in touch_files:
                             f.touch()
+                    if len(non_existing_files) > 0:
+                        job.log_warning(
+                            msg=f"Missing rule's output files: "
+                            + f"{', '.join(non_existing_files)}. --touch skipped for this job."
+                        )
```

Comment:
```suggestion
                        logger.warning(
                            f"Output files not touched because they don't exist: {', '.join(non_existing_files)}"
                        )
```